app/src/androidTest/java/com/github/houseorganizer/houseorganizer/CreateHouseholdActivityTest.java

```diff
@@ -69,19 +81,64 @@ public void setupHouseholds() throws ExecutionException, InterruptedException {
 
         Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
         context.sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
+        createHouseholdRule.getScenario().onActivity(a -> decorView = a.getWindow().getDecorView());
     }
 
     @Test
     public void createHouseholdWorks() throws ExecutionException, InterruptedException {
+
+        final String houseName = "MyHouse";
+
         onView(withId(R.id.editTextHouseholdName)).perform(click(),
-                typeText("Test"), closeSoftKeyboard());
-        onView(withId(R.id.editTextLatitude)).perform(click(),
-                typeText("45"), closeSoftKeyboard());
-        onView(withId(R.id.editTextLongitude)).perform(click(),
-                typeText("45"), closeSoftKeyboard());
+                typeText(houseName), closeSoftKeyboard());
+        onView(withId(R.id.editTextAddress)).perform(
+                typeText("EPFL, Lausanne"), closeSoftKeyboard());
         onView(withId(R.id.submitHouseholdButton)).perform(click());
+        Thread.sleep(4000);
+        onView(withText(R.string.confirm)).perform(click());
+        Thread.sleep(1000);
```

Comment:
@sansive might want to get rid of this thread sleeps

Reply:
the first one is special as we have to wait for the location to be retrieved but we can see if @sansive has a solution

Reply:
Thanks for the mention, I'll look into it